### PR TITLE
Configuration-as-Code compatibility

### LINKED ISSUE #20: Configuration-as-Code compatibility
Add full support for the [Jenkins Configuration as Code Plugin](https://github.com/jenkinsci/configuration-as-code-plugin). This mainly includes the global configurations for ECU-TEST installations and TEST-GUIDE servers.

Example configurations:

**ECU-TEST**
```
tool:
  ecu-test:
    installations:
    - home: "C:\\Program Files\\ECU-TEST 7.2"
      name: "ECU-TEST 7.2"
```

**TEST-GUIDE**
```
unclassified:
  test-guide:
    installations:
      - config:
        settings:
          - atx-boolean-setting:
            name: "uploadToServer"
            value: true
          - atx-text-setting:
            name: "serverURL"
            value: "127.0.0.1"
          - ...
        customSettings:
          - atx-custom-boolean-setting:
            name: "customOption"
            checked: true
          - atx-custom-text-setting:
            name: "customLabel"
            value: "test"
          - ...
      name: "TEST-GUIDE"
      toolName: "ECU-TEST"
```

Tracking: https://github.com/jenkinsci/configuration-as-code-plugin/issues/809

### PR DESCRIPTION
**Pull request by cpoenisch**
Resolves #20

### CODE REVIEW

**Comment by cpoenisch** [comment]
Tracking: https://github.com/jenkinsci/configuration-as-code-plugin/issues/809

**Comment by casz** [comment]
Thanks for using the GitHub tracking issue. Means I can avoid JIRA :blush:

File: src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ConfigurationAsCodeIT.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @author Christian Pönisch <christian.poenisch@tracetronic.de>
+ */
+public class ConfigurationAsCodeIT extends IntegrationTestBase {
+
+    @Test
+    public void testImportConfiguration() throws Exception {
+        final URL yamlConfig = this.getClass().getResource("configuration-as-code.yml");
+        System.out.println(yamlConfig);
+        ConfigurationAsCode.get().configure(yamlConfig.toString());
```
Comment (casz) [inline src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ConfigurationAsCodeIT.java]: Consider using the annotation `@ConfiguredWithCode("configuration-as-code.yml")` which the `tests` classifier includes.

**Comment by casz** [comment]
Tests are looking good! A lot cleaner now that you are not using the YAML load method :sweat_smile:

**Comment by cpoenisch** [comment]
Yep thanks for your hints!

File: src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ETConfigurationAsCodeIT.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        assertThat(((ETToolProperty) installation.getProperties().get(0)).getTimeout(), is(60));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void testExportConfiguration() throws Exception {
+        final ETInstallation.DescriptorImpl etDescriptor = jenkins.jenkins
+            .getDescriptorByType(ETInstallation.DescriptorImpl.class);
+        etDescriptor.setInstallations(new ETInstallation("ECU-TEST", "C:\\ECU-TEST",
```
Comment (casz) [inline src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ETConfigurationAsCodeIT.java]: Not sure you need to override the configuration, could validate the config loaded in from `@ConfiguredWithCode("configuration-as-code.yml")` :see_no_evil:

Comment (cpoenisch) [reply src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ETConfigurationAsCodeIT.java]: You're absolutely right! I'll fix it.

Comment (casz) [reply src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ETConfigurationAsCodeIT.java]: Great now your fully validating the YAML config :sheep:

**Pull request merged**